### PR TITLE
feat: support custom SSL certs in SQL lambda handler

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

Adds support for custom SSL certs in SQL data sources. Changes:

- Updates CDK Construct API to add an optional `sslCertConfig` attribute to the SQL data source configs
- Updates the Lambda handler deployed to the customer account to look up the SSM paths and return the custom SSL certificate if found
- Adds beginnings of support for Beta SQL Layer in CDK test

#### New CDK parameters & API

cc: @atierian, @swaminator 

AmplifyGraphqlApi and AmplifyData constructs add a new `sslCertConfig` parameter to the SQL db configs. New config API:

**Type declarations**
```ts
// NEW
export interface SslCertConfig {}

// NEW
export interface SslCertSsmPathConfig extends SslCertConfig {
  readonly ssmPath: string | string[];
}

// MODIFIED
export interface SqlModelDataSourceSsmDbConnectionStringConfig {
  // ...
  // V--- Added
  readonly sslCertConfig?: SslCertConfig;
}

// MODIFIED
export interface SqlModelDataSourceSecretsManagerDbConnectionConfig {
  // ...
  // V--- Added
  readonly sslCertConfig?: SslCertConfig;
}

// MODIFIED
export interface SqlModelDataSourceSsmDbConnectionConfig {
  // ...
  // V--- Added
  readonly sslCertConfig?: SslCertConfig;
}
```

**Call site**
```typescript
const api = new AmplifyGraphqlApi(stack, 'MyApi', {
  definition: AmplifyGraphqlDefinition.fromString(schema, {
    name: 'myStrategy',
    dbType: 'MYSQL',
    dbConnectionConfig: {
      connectionUriSsmPath: '/ssm/path/to/connectionUri',
      // V-------- THIS IS NEWLY ADDED
      sslCertConfig: {
        ssmPath: ['/ssm/path/to/sslCert/1', '/ssm/path/to/sslCert/2']
      }
    }

  }),
  authorizationModes,
});
```

Note that we should be able to provide a more streamlined interface in the data schema builder; the additional structure in the CDK construct is required to allow disjoint-union-type behavior in a JSII-compatible interface.

#### Description of how you validated changes

- Unit tests
- E2E tests. Ideally, we'd set up a database that uses a totally custom cert for validation, but Aurora doesn't support custom SSL. Instead, we'll validate that the handler & layer use the SSL cert we pass:
  - Test 1 specifies an invalid SSL cert, and is expected to fail attempting to connect to the database
  - Test 2 specifies a valid SSL cert that duplicates the RDS bundle, and is expected to succeed

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] E2Es pass
  - [x] [All but custom SSL](https://us-east-1.console.aws.amazon.com/codesuite/codebuild/594813022831/projects/amplify-category-api-e2e-workflow/batch/amplify-category-api-e2e-workflow:65ea70a9-8dba-40a8-93ce-b2fbaf9454f1?region=us-east-1)
  - [x] [Custom SSL](https://us-east-1.console.aws.amazon.com/codesuite/codebuild/594813022831/projects/amplify-category-api-e2e-workflow/batch/amplify-category-api-e2e-workflow:8fe9a89a-eae8-4c2f-b232-80755de1f4e9?region=us-east-1)
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-api-construct/API.md [hunk 1]
```diff
@@ -364,7 +364,7 @@ export class SQLLambdaModelDataSourceStrategyFactory {
 export type SqlModelDataSourceDbConnectionConfig = SqlModelDataSourceSecretsManagerDbConnectionConfig | SqlModelDataSourceSsmDbConnectionConfig | SqlModelDataSourceSsmDbConnectionStringConfig;
 
 // @public
-export interface SqlModelDataSourceSecretsManagerDbConnectionConfig {
+export interface SqlModelDataSourceSecretsManagerDbConnectionConfig extends SslCertSsmPathAwareConfig {
```
Comment (sundersc) [inline packages/amplify-graphql-api-construct/API.md:367]: This is not ideal. When using secrets manager for credentials, I think we shouldn't use (mix) SSM for SSL certificate alone.

Comment (sundersc) [reply packages/amplify-graphql-api-construct/API.md:367]: We can 
(1) for now, support ssl certificate only when using SSM config (which is enough to support Gen2 DX).
(2) later, we will support secret manager for SSL certificate.

Comment (palpatim) [reply packages/amplify-graphql-api-construct/API.md:367]: Discussed offline. Will remove from Secrets Manager

**Review by atierian — DISMISSED** [review]
API BR approved.
(I did not review the implementation)

**Review by sundersc — DISMISSED** [review]

**Review by phani-srikar — DISMISSED** [review]

**Review by svidgen — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

**Comment by palpatim** [comment]
Reviewed offline with @swaminator and @atierian, both approved API changes.

**Pull request merged**